### DISCUSSION TITLE
How to programmatically scroll a ScrollArea to the end?

### DISCUSSION THREAD

**Discussion by singalen** [emilk/egui#951] [Q&A] [answered]
Looking at `Prepared::end()` code, I don't see an obvious API to do it.
Even if I find a way to set `State.offset`, what value to set it to?

**Comment by BctfN0HUK7Yg** [comment, marked as answer]
look a to the demo https://github.com/emilk/egui/blob/master/egui_demo_lib/src/apps/demo/scrolling.rs

if scroll_bottom {
                ui.scroll_to_cursor(Align::BOTTOM);
            }

**Comment by singalen** [comment]
Thank you! My bad, I missed it.
Now I get it that it's the `Ui` inside a `ScrollArea` needs to be scrolled.
